workflows_core/dataset/field.py

```diff
@@ -189,58 +189,64 @@ def insert_centroids(self, centroid_documents: DocumentList, alias: str):
         return self._dataset.api._insert_centroids(
             dataset_id=self.dataset_id,
             cluster_centers=centroid_documents,
-            vector_fields=[self.field],
+            vector_fields=[self._text_field],
             alias=alias,
         )
 
     def get_centroids(self, alias: str):
         return self._dataset.api._get_centroids(
             dataset_id=self.dataset_id,
-            vector_fields=[self.field],
+            vector_fields=[self._text_field],
             alias=alias,
         )
 
 
 class KeyphraseField(Field):
     def __init__(self, dataset, field: str):
         super().__init__(dataset=dataset, field=field)
+        _, text_field, alias, *_ = field.split(".")
+        self._text_field_ = text_field
+        self._alias_ = alias
```

Comment:
Why do we have an ending underscore?

Reply:
cos the attr is already defined in `Field`. These need to be separate to track the split fields

Reply:
`_keyphrase_alias` is better than `_alias_` because its more descriptive about the difference

Reply:
fixed